File: backend/app/schemas/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.document import DocumentStatus


class DocumentUploadResponse(BaseModel):
    """Returned immediately by POST /api/documents/upload."""
    id: uuid.UUID
    filename: str
    status: DocumentStatus


class DocumentListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    filename: str
    file_type: str
    file_size: int
    status: DocumentStatus
    page_count: int
    chunk_count: int
    created_at: datetime
    updated_at: datetime


class DocumentDetail(DocumentListItem):
    error_message: str | None = None

File: frontend/src/types/document.ts
export type DocumentStatus = "uploaded" | "processing" | "processed" | "failed";

export interface DocumentUploadResponse {
  id: string;
  filename: string;
  status: DocumentStatus;
}

export interface DocumentListItem {
  id: string;
  filename: string;
  file_type: string;
  file_size: number;
  status: DocumentStatus;
  page_count: number;
  chunk_count: number;
  created_at: string;
  updated_at: string;
}

export interface DocumentDetail extends DocumentListItem {
  error_message: string | null;
}
